src/Bicep.Core/Analyzers/Linter/LinterRuleBase.cs

```diff
@@ -0,0 +1,162 @@
+        {
+            return (values.Any() ? FormatMessage(values) : this.Description)
+                + Environment.NewLine
+                + string.Format(CoreResources.SeeDocLinkFormat, this.DocumentationUri);
+        }
+
+        public IEnumerable<IBicepAnalyzerDiagnostic> Analyze(SemanticModel model)
+        {
+            try
+            {
+                // Expand the iteration immediately or the try/catch won't catch exceptions occuring during the rule analysis
+                // TODO: We need exception handling further up the tree in order to handle external linters.
+                return AnalyzeInternal(model).ToArray();
+            }
+            catch (Exception ex)
+            {
+                return new[]{ new AnalyzerDiagnostic(this.AnalyzerName,
+                                                    new TextSpan(0, 0),
+                                                    DiagnosticLevel.Warning,
+                                                    FailedRuleCode,
+                                                    string.Format(CoreResources.LinterRuleExceptionMessageFormat, ex.Message),
+                                                    null)
+                };
+            }
+        }
+
+        /// <summary>
+        /// Abstract method each rule must implement to provide analyzer
+        /// diagnostics through the Analyze API
+        /// </summary>
+        /// <param name="model"></param>
+        /// <returns></returns>
+        internal abstract IEnumerable<IBicepAnalyzerDiagnostic> AnalyzeInternal(SemanticModel model);
+
+        /// <summary>
+        /// Get a setting from defaults or local override
+        /// Expectation: key names for settings are lower case
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="name"></param>
+        /// <param name="defaultValue"></param>
+        /// <returns></returns>
+        protected T GetConfiguration<T>(string name, T defaultValue)
+            => ConfigurationBinder.GetValue(this.Config, $"{RuleConfigSection}:{Code}:{name}", defaultValue);
+
+        /// <summary>
+        /// Get a section of the config file as an array of strings.
+        /// Expectation: all key names shoult be lower case
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="name"></param>
+        /// <param name="defaultValue"></param>
+        /// <returns></returns>
+        protected T[] GetArray<T>(string name, T[] defaultValue)
+            => this.Config?.GetSection($"{RuleConfigSection}:{Code}:{name.ToLower()}").Get<T[]>() ?? defaultValue;
+
+        /// <summary>
+        ///  Create a simple diagnostic that displays the defined Description
+        ///  of the derived rule.
+        /// </summary>
+        /// <param name="span"></param>
+        /// <returns></returns>
+        internal virtual AnalyzerDiagnostic CreateDiagnosticForSpan(TextSpan span) =>
```

Comment:
>internal [](http://example.com/codeflow?start=8&length=8)

Shouldn't these be protected so the derived classes can all them regardless of which assembly they're in?